Bug: Cannot convert bpmn:error

### Describe the bug
```
org.camunda.community.migration.converter.exception.VisitorException: Exception in visitor org.camunda.community.migration.converter.visitor.impl.eventReference.ErrorVisitor while visiting element  -> bpmn:definitions (Definitions_1qyhhxm) -> bpmn:error (Error_1uirwzv) caused by: Cannot invoke "org.camunda.community.migration.converter.expression.ExpressionTransformationResult.getNewExpression()" because "expressionTransformationResult" is null
    at org.camunda.community.migration.converter.visitor.AbstractFilteringVisitor.visit(AbstractFilteringVisitor.java:20)
    at org.camunda.community.migration.converter.BpmnConverter.lambda$traverse$12(BpmnConverter.java:190)
    at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
    at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(Unknown Source)
    at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
    at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
    at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
    at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
    at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
    at java.base/java.util.stream.ReferencePipeline.forEach(Unknown Source)
    at org.camunda.community.migration.converter.BpmnConverter.traverse(BpmnConverter.java:190)
    at org.camunda.community.migration.converter.BpmnConverter.lambda$traverse$13(BpmnConverter.java:191)
    at java.base/java.util.ArrayList.forEach(Unknown Source)
    at org.camunda.community.migration.converter.BpmnConverter.traverse(BpmnConverter.java:191)
    at org.camunda.community.migration.converter.BpmnConverter.check(BpmnConverter.java:66)
    at org.camunda.community.migration.converter.webapp.BpmnConverterService.check(BpmnConverterService.java:48)
    at org.camunda.community.migration.converter.webapp.ConverterController.check(ConverterController.java:50)
    at jdk.internal.reflect.GeneratedMethodAccessor13.invoke(Unknown Source)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.base/java.lang.reflect.Method.invoke(Unknown Source)
    at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
    at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
    at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)
    at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)
    at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
    at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
    at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
    at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
    at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
    at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
    at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
    at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
    at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
    at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
    at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
    at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:768)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
    at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:389)
    at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
    at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:926)
    at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
    at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
    at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
    at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
    at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
    at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "org.camunda.community.migration.converter.expression.ExpressionTransformationResult.getNewExpression()" because "expressionTransformationResult" is null
    at org.camunda.community.migration.converter.visitor.impl.eventReference.ErrorVisitor.postCreationVisitor(ErrorVisitor.java:35)
    at org.camunda.community.migration.converter.visitor.AbstractProcessElementVisitor.visitBpmnElement(AbstractProcessElementVisitor.java:10)
    at org.camunda.community.migration.converter.visitor.AbstractBpmnElementVisitor.visitFilteredElement(AbstractBpmnElementVisitor.java:18)
    at org.camunda.community.migration.converter.visitor.AbstractFilteringVisitor.visit(AbstractFilteringVisitor.java:16)
    ... 66 more
```
### Steps to reproduce

1. Create a diagram containing an error event with an empty error code
2. Convert the diagram
3. The exception occurs

### Expected Behaviour

On empty error code, nothing special should happen

### Library version

0.6.1

### Camunda version

*

--- a/backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/conversion/ErrorConversion.java
+++ b/backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/conversion/ErrorConversion.java
@@ -12,6 +12,8 @@ public class ErrorConversion extends AbstractTypedConversion<ErrorConvertible> {
 
   @Override
   protected void convertTyped(DomElement element, ErrorConvertible convertible) {
-    element.setAttribute(NamespaceUri.BPMN, "errorCode", convertible.getErrorCode());
+    if (convertible.getErrorCode() != null) {
+      element.setAttribute(NamespaceUri.BPMN, "errorCode", convertible.getErrorCode());
+    }
   }
 }

--- a/backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/visitor/impl/eventReference/ErrorVisitor.java
+++ b/backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/visitor/impl/eventReference/ErrorVisitor.java
@@ -30,6 +30,9 @@ public class ErrorVisitor extends AbstractEventReferenceVisitor {
   @Override
   protected void postCreationVisitor(DomElementVisitorContext context) {
     String errorCode = context.getElement().getAttribute(BPMN, "errorCode");
+    if (errorCode == null) {
+      return;
+    }
     ExpressionTransformationResult expressionTransformationResult =
         ExpressionTransformer.transform(errorCode);
     if (expressionTransformationResult.getNewExpression().startsWith("=")) {

--- a/backend-diagram-converter/core/src/test/java/org/camunda/community/migration/converter/BpmnConverterTest.java
+++ b/backend-diagram-converter/core/src/test/java/org/camunda/community/migration/converter/BpmnConverterTest.java
@@ -399,9 +399,13 @@ public class BpmnConverterTest {
     assertThat(
             modelInstance.getDocument().getElementById("Error_16zktjx").getAttribute("errorCode"))
         .isEqualTo("someCode");
-    BpmnConverter converter = BpmnConverterFactory.getInstance().get();
-    StringWriter writer = new StringWriter();
-    converter.printXml(modelInstance.getDocument(), true, writer);
-    System.out.println(writer.toString());
+  }
+
+  @Test
+  void testErrorCodeMayBeNull() {
+    BpmnModelInstance modelInstance = loadAndConvert("null-error-code.bpmn");
+    assertThat(
+            modelInstance.getDocument().getElementById("Error_16zktjx").getAttribute("errorCode"))
+        .isNull();
   }
 }

--- a/backend-diagram-converter/core/src/test/resources/null-error-code.bpmn
+++ b/backend-diagram-converter/core/src/test/resources/null-error-code.bpmn
@@ -1,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:modeler="http://camunda.org/schema/modeler/1.0" id="Definitions_1gbx4qr" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.12.1" modeler:executionPlatform="Camunda Platform" modeler:executionPlatformVersion="7.19.0">
+  <bpmn:process id="TransactionTestProcess" name="Transaction Test" isExecutable="true">
+    <bpmn:startEvent id="Event_1i3htun" name="Error code conversion should be tested">
+      <bpmn:outgoing>Flow_0we8gbc</bpmn:outgoing>
+    </bpmn:startEvent>
+    <bpmn:sequenceFlow id="Flow_0we8gbc" sourceRef="Event_1i3htun" targetRef="Activity_1h3hatw" />
+    <bpmn:serviceTask id="Activity_1h3hatw" name="Throw error">
+      <bpmn:incoming>Flow_0we8gbc</bpmn:incoming>
+      <bpmn:outgoing>Flow_0y3k84c</bpmn:outgoing>
+    </bpmn:serviceTask>
+    <bpmn:endEvent id="Event_09a497u" name="Error code conversion done">
+      <bpmn:incoming>Flow_0y3k84c</bpmn:incoming>
+    </bpmn:endEvent>
+    <bpmn:sequenceFlow id="Flow_0y3k84c" sourceRef="Activity_1h3hatw" targetRef="Event_09a497u" />
+    <bpmn:boundaryEvent id="Event_0apc6gw" name="Error thrown" attachedToRef="Activity_1h3hatw">
+      <bpmn:outgoing>Flow_03nn2dd</bpmn:outgoing>
+      <bpmn:errorEventDefinition id="ErrorEventDefinition_1ue91kl" errorRef="Error_16zktjx" />
+    </bpmn:boundaryEvent>
+    <bpmn:endEvent id="Event_1ey9c8f" name="Error handled">
+      <bpmn:incoming>Flow_03nn2dd</bpmn:incoming>
+    </bpmn:endEvent>
+    <bpmn:sequenceFlow id="Flow_03nn2dd" sourceRef="Event_0apc6gw" targetRef="Event_1ey9c8f" />
+  </bpmn:process>
+  <bpmn:error id="Error_16zktjx" name="SomeName" />
+  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
+    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="TransactionTestProcess">
+      <bpmndi:BPMNShape id="Event_1i3htun_di" bpmnElement="Event_1i3htun">
+        <dc:Bounds x="142" y="102" width="36" height="36" />
+        <bpmndi:BPMNLabel>
+          <dc:Bounds x="116" y="145" width="89" height="40" />
+        </bpmndi:BPMNLabel>
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Activity_10af0py_di" bpmnElement="Activity_1h3hatw">
+        <dc:Bounds x="230" y="80" width="100" height="80" />
+        <bpmndi:BPMNLabel />
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Event_09a497u_di" bpmnElement="Event_09a497u">
+        <dc:Bounds x="382" y="102" width="36" height="36" />
+        <bpmndi:BPMNLabel>
+          <dc:Bounds x="360" y="145" width="81" height="27" />
+        </bpmndi:BPMNLabel>
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Event_1ey9c8f_di" bpmnElement="Event_1ey9c8f">
+        <dc:Bounds x="362" y="222" width="36" height="36" />
+        <bpmndi:BPMNLabel>
+          <dc:Bounds x="347" y="265" width="67" height="14" />
+        </bpmndi:BPMNLabel>
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNShape id="Event_0k38m8v_di" bpmnElement="Event_0apc6gw">
+        <dc:Bounds x="272" y="142" width="36" height="36" />
+        <bpmndi:BPMNLabel>
+          <dc:Bounds x="260" y="185" width="61" height="14" />
+        </bpmndi:BPMNLabel>
+      </bpmndi:BPMNShape>
+      <bpmndi:BPMNEdge id="Flow_0we8gbc_di" bpmnElement="Flow_0we8gbc">
+        <di:waypoint x="178" y="120" />
+        <di:waypoint x="230" y="120" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_0y3k84c_di" bpmnElement="Flow_0y3k84c">
+        <di:waypoint x="330" y="120" />
+        <di:waypoint x="382" y="120" />
+      </bpmndi:BPMNEdge>
+      <bpmndi:BPMNEdge id="Flow_03nn2dd_di" bpmnElement="Flow_03nn2dd">
+        <di:waypoint x="290" y="178" />
+        <di:waypoint x="290" y="240" />
+        <di:waypoint x="362" y="240" />
+      </bpmndi:BPMNEdge>
+    </bpmndi:BPMNPlane>
+  </bpmndi:BPMNDiagram>
+</bpmn:definitions>

